homeassistant/components/zwave_js/api.py

```diff
@@ -571,6 +574,104 @@ def node_removed(event: dict) -> None:
     )
 
 
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/begin_healing_network",
+        vol.Required(ENTRY_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_get_entry
+async def websocket_begin_healing_network(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Begin healing the Z-Wave network."""
+    controller = client.driver.controller
+
+    @callback
+    def async_cleanup() -> None:
+        """Remove signal listeners."""
+        for unsub in unsubs:
+            unsub()
+
+    @callback
+    def forward_event(key: str, event: dict) -> None:
+        connection.send_message(
+            websocket_api.event_message(
+                msg[ID], {"event": event["event"], "data": event[key]}
+            )
+        )
+
+    connection.subscriptions[msg["id"]] = async_cleanup
+    unsubs = [
+        controller.on("heal network progress", partial(forward_event, "progress")),
+        controller.on("heal network done", partial(forward_event, "result")),
+    ]
+
+    result = await controller.async_begin_healing_network()
+    connection.send_result(
+        msg[ID],
+        result,
+    )
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/stop_healing_network",
+        vol.Required(ENTRY_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_get_entry
+async def websocket_stop_healing_network(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Stop healing the Z-Wave network."""
+    controller = client.driver.controller
+    result = await controller.async_stop_healing_network()
```

Comment:
When sending this, we won't get the `heal network done` event and so the above subscription might never be undone ?

Reply:
Do we want to be "smart" and automatically call stop healing network when we unsubscribe from the `start_healing_network` command and the healing is not done yet ?

Reply:
Hmm... Do we want network healing to stop because the user leaves the page? My understanding is that would trigger an unsubscription. My thought was that the page could have a stop network healing button that could both call this endpoint and stop the subscription.

That brings up another point though. I don't know if there's a way to know the current network healing status, so if I leave the page and the healing is still going on, then come back, how will the UI know to resubscribe to updates? I think I might need to do more research on this before we move forward. If I can figure that one out, I may need to add a separate WS command to subscribe to updates.

Reply:
good point that we don't want to stop it at the end of the sub and on checking if healing is ongoing. We could store it locally, since we know when we send a start?

Reply:
I submitted a PR to add a controller property that will tell you if healing is still going on (merged earlier today). I will add a server command to check this (similar to is_statistics_enabled) and then we can add a WS API command for that. We should then split out the event subscriptions into a separate command. This will have to stay on hold until the next zwave-js release

Reply:
https://github.com/zwave-js/zwave-js-server/pull/246